github_pr.py

```diff
@@ -108,12 +111,70 @@ def github_list_prs(**args):
     elif 'head' in args and args['head']:
         prs = _load_prs_by_branch(**args)
         _print_prs(prs, **args)
+    elif 'filters' in args and args['filters']:
+        prs = github_filter_prs(**args)
+        _print_prs(prs, **args)
     else:
         prs = repo.get_pulls()
         _print_prs(prs, **args)
     return list_return_obj
 
 
+def github_filter_prs(**args):
+    """
+    Filters prs to return only what is contained in the filters
+    """
+    git_hub = Github(args['token'])
+    repo = git_hub.get_repo(args['repo'])
+    all_prs = [{'pr':repo.get_pull(pr.number), 'issue':repo.get_issue(pr.number)} for pr in repo.get_pulls()]
+    filters = {filter.split('=')[0]:filter.split('=')[1] for filter in args['filters'].split(',')}
+
+    if 'owner' in filters:
+        all_prs = return_specific_owner_prs(all_prs, filters)
+    if 'label' in filters:
+        all_prs = return_specific_labeled_prs(all_prs, filters)
+    if 'status' in filters:
+        all_prs = return_specific_status_prs(all_prs, filters)
+    if 'comment' in filters:
+        all_prs = return_specific_comment_prs(all_prs, filters)
+
+    return all_prs
+
+
+def return_specific_owner_prs(all_prs, filters):
```

Comment:
these 4 functions are very similar. Can you abstract this in a way where there is less duplication in the logic?